code/game/objects/items/storage/fancy.dm

```diff
@@ -81,9 +82,34 @@
 	for(var/i in 1 to 6)
 		new /obj/item/reagent_containers/food/snacks/donut(src)
 
+/obj/item/storage/fancy/donut_box/PopulateContents()
+	. = ..()
+	update_icon()
+
+/obj/item/storage/fancy/donut_box/update_icon_state()
+	. = ..()
+	if(fancy_open)
+		icon_state = "donutbox_inner"
+	else
+		icon_state = "donutbox"
+
+/obj/item/storage/fancy/donut_box/update_overlays()
+	. = ..()
+
+	if (!fancy_open)
+		return
+
+	var/donuts = 0
+
+	for (var/obj/item/reagent_containers/food/snacks/donut/donut in contents)
+
+		. += image(icon = initial(icon), icon_state = donut.in_box_sprite(), pixel_x = donuts * DONUT_INBOX_SPRITE_WIDTH)
+		donuts += 1
+
+	. += image(icon = initial(icon), icon_state = "donutbox_top")
+
 /obj/item/storage/fancy/donut_box/deadly
 	icon = 'icons/obj/food/containers.dmi'
-	icon_state = "donutbox6"
 	icon_type = "donut"
 	name = "donut box"
 	spawn_type = /obj/item/reagent_containers/food/snacks/donut/deadly
```

Comment:
The only thing that the deadly donut box should have is the defines on its ``spawn_type``. Everything else is copy pasted.